### PR TITLE
feat: Spanner add tests for FGAC Postgres dialect

### PR DESCRIPTION
**Pull request by hemanshv**
Adds integration tests for FGAC pg.

### CODE REVIEW

**Comment by hemanshv** [comment]
Hi @jskeet , @amanda-tarafa ,
Please review this PR and share your valuable feedback.

Thanks,

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/SpannerReaderRoleFixturePg.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.Spanner.Data.CommonTesting;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests;
+
+[CollectionDefinition(nameof(SpannerReaderRoleFixturePg))]
+public class SpannerReaderRoleFixturePg : SpannerTableFixturePostgreSql, ICollectionFixture<SpannerReaderRoleFixturePg>
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/SpannerReaderRoleFixturePg.cs:21]: Our other Postgres fixtures just have a suffix of PostgreSql - so I'd expect this to be `SpannerReaderRoleFixturePostgreSql`

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/SpannerReaderRoleFixturePg.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+{
+    public const string ReaderDatabaseRole = "PgReader";
+
+    public SpannerReaderRoleFixturePg(): base("PgDbRoleTable")
+    {
+    }
+    protected override void CreateTable()
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/SpannerReaderRoleFixturePg.cs:28]: Nit: add blank line.

**Review by jskeet — COMMENTED** [review]
It's not clear to me whether this integration test is really required, although it basically looks okay. I'm nervous that we end up creating a lot of duplication which could become inconsistent, if we gradually add coverage for all integration tests.

Basically there's nothing "wrong" here (beyond the comments in the review) but I'm vaguely uncomfortable...

@amanda-tarafa can probably express this better than I can - she tends to do so when it comes to testing, particularly.

**Comment by anuragsrivstv** [comment]
Do we have any related library change also ? If not then may be we should not write integration tests for it ?
Also is this functionality not covered in samples ?

**Review by amanda-tarafa — COMMENTED** [review]
In the past [we've agreed](https://github.com/googleapis/google-cloud-dotnet/pull/9164#pullrequestreview-1125115366) that we would only test features in Postgres dialect when there was Postgres specific library code. That's not the case here, and we are already [testing granular access for the Google dialect](https://github.com/googleapis/google-cloud-dotnet/blob/main/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/SpannerGranularAccessTest.cs).

In my opinion this PR should be closed.

**Pull request closed without merging**